content/src/denylist/ActiveDenylist.ts

```diff
@@ -0,0 +1,149 @@
+import { EthAddress } from 'dcl-crypto'
+import { validateSignature } from 'decentraland-katalyst-commons/signatures'
+import { DenylistRepository } from '../repository/extensions/DenylistRepository'
+import { Repository } from '../repository/Repository'
+import { DB_REQUEST_PRIORITY } from '../repository/RepositoryQueue'
+import { ContentAuthenticator } from '../service/auth/Authenticator'
+import { ContentCluster } from '../service/synchronization/ContentCluster'
+import {
+  Denylist,
+  DenylistAction,
+  DenylistMetadata,
+  DenylistSignatureValidationResult,
+  DenylistSignatureValidationStatus,
+  isErrorOperation
+} from './Denylist'
+import { DenylistTarget, DenylistTargetId, DenylistTargetType } from './DenylistTarget'
+
+export class ActiveDenylist extends Denylist {
+  private size: number = 0
+  constructor(
+    private readonly repository: Repository,
+    private readonly authenticator: ContentAuthenticator,
+    private readonly cluster: ContentCluster,
+    private readonly network: string
+  ) {
+    super()
+  }
+
+  async addTarget(target: DenylistTarget, metadata: DenylistMetadata): Promise<DenylistSignatureValidationResult> {
+    // Validate blocker and signature
+    const operationResult: DenylistSignatureValidationResult = await this.validateSignature(
+      DenylistAction.ADDITION,
+      target,
+      metadata
+    )
+    if (isErrorOperation(operationResult)) {
+      return operationResult
+    }
+
+    await this.repository.tx(
+      async (transaction) => {
+        // Add denylist
+        await transaction.denylist.addTarget(target)
+
+        // Add to history
+        await transaction.denylist.addEventToHistory(target, metadata, DenylistAction.ADDITION)
+      },
+      { priority: DB_REQUEST_PRIORITY.HIGH }
+    )
+    this.size++
+    return { status: DenylistSignatureValidationStatus.OK }
+  }
+
+  async removeTarget(target: DenylistTarget, metadata: DenylistMetadata): Promise<DenylistSignatureValidationResult> {
+    // Validate blocker and signature
+    const operationResult: DenylistSignatureValidationResult = await this.validateSignature(
+      DenylistAction.REMOVAL,
+      target,
+      metadata
+    )
+    if (isErrorOperation(operationResult)) {
+      return operationResult
+    }
+
+    await this.repository.tx(
+      async (transaction) => {
+        // Remove denylist
+        await transaction.denylist.removeTarget(target)
+
+        // Add to history
+        await transaction.denylist.addEventToHistory(target, metadata, DenylistAction.REMOVAL)
+      },
+      { priority: DB_REQUEST_PRIORITY.HIGH }
+    )
+    this.size--
```

Comment:
Writing it here, but have the same concern for `addTarget`. What happens if you remove the same target many times? Does it fail the second time? If not, then this might not be the best way to keep track of whether there is something on the denylist